src/main/java/ai/tecton/client/TectonClient.java

```diff
@@ -188,4 +188,85 @@ private HttpResponse getHttpResponse(AbstractTectonRequest tectonRequest) {
     }
     return httpResponse;
   }
+
+  /**
+   * A Builder class for creating an instance of {@link TectonClient} object with specific
+   * configurations
+   */
+  public static class Builder {
+    private String url;
+    private String apiKey;
+    private TectonClientOptions tectonClientOptions;
+    private OkHttpClient httpClient;
+
+    /**
+     * Setter for url
+     *
+     * @param url The Tecton Base Url
+     * @return this Builder
+     */
+    public Builder url(String url) {
```

Comment:
Id check but does the kotlin assignment operator need a specific method name? Like 'Builder.setX' which kotlin can then translate to 'builder.X = something'?

Reply:
Tested in a Kotlin Application and this works